### ISSUE TITLE
Capture and process signals such as system interrupts

### ISSUE THREAD

**Issue by JinnyYi** [beyondstorage/beyond-ctl#112]
Support recieving and handling `ctrl` + `c` and exiting application.

**Comment by JinnyYi** [comment]
Do we need to clean up any unfinished data before exiting the application, such as abort multipart upload?

**Comment by Xuanwo** [comment]
> Do we need to clean up any unfinished data before exiting the application, such as abort multipart upload?

No, while we receive a `Ctrl+C`, we should exit as soon as possible.
